lib/core/security/tokenRepository.ts

```diff
@@ -352,6 +387,38 @@ export class TokenRepository extends ObjectRepository<Token> {
     return userToken;
   }
 
+  async _verifyApiKey(decoded, token: string) {
+    const fingerprint = sha256(token);
+
+    const userApiKeys = await ApiKey.search({
+      query: {
+        term: {
+          userId: decoded._id,
```

Comment:
Why not searching by `fingerprint` first and then checking that the `userID` matches?
Then the search would only return 0 or 1 result, and it would save the `Array.find` call.

Reply:
Because it lead to introduce a breaking (fingerprint is not indexed atm and changes in mappings are not taken in account at start)

Reply:
Good call! Nevermind then.